VirtualH89/Src/computer.h

```diff
@@ -10,8 +10,8 @@
 #define COMPUTER_H_
 
 #include "h89Types.h"
+#include "AddressBus.h"
```

Comment:
I was wondering about those forward declarations. So that is intentional? I've not seen C++ code done that way. Usually the header file defines the interface to the object and as such all users of the object are expected to include the header. Among other things, it gives the compiler more opportunities to optimize.

If you are firmly committed to that design philosophy, obviously I'll have to change the code. There are some places where fwd declare will not work as-is, and I'll have to rework the code for that. Can you explain more about the benefits of avoiding including the headers?

I'll address the other problems either way.
Thanks,
Doug

Sent from my Samsung Galaxy Tab®|PRO

-------- Original message --------
From: Mark Garlanger <notifications@github.com> 
Date:02/20/2016  10:49  (GMT-06:00) 
To: mgarlanger/VirtualH89 <VirtualH89@noreply.github.com> 
Cc: durgadas311 <durgadas311@gmail.com> 
Subject: Re: [VirtualH89] A large set of changes to support MMS77316 floppy
  disk controller and 77320 SASI ctrlr (#7) 

In VirtualH89/Src/computer.h:

> @@ -10,8 +10,8 @@
>  #define COMPUTER_H_
>  
>  #include "h89Types.h"
> +#include "AddressBus.h"
Was there a reason to include this, instead of keeping it forward declared?

—
Reply to this email directly or view it on GitHub.